qiskit_experiments/framework/json.py

```diff
@@ -455,6 +456,27 @@ def default(self, obj: Any) -> Any:  # pylint: disable=arguments-differ
             return _serialize_bytes(obj)
         if dataclasses.is_dataclass(obj):
             return _serialize_object(obj, settings=dataclasses.asdict(obj))
+        if isinstance(obj, UFloat):
```

Comment:
Unfortunately this is not possible because `Variable` has hash function based on object id. Indeed we can serialize `AffineScalarFunc` with correlation information, i.e. dict of `Variable` with correlation parameter, but deserialized `Value` object is no longer equivalent due to different memory allocation. This is why I tried to monkey patch this object so that it can use uuid instead of object id.

Reply:
What I meant was if you wanted to just save a list of correlated variables `[a, b, c,...]` in principle you could save the list of nominal values + the correlation matrix and recover them via `a,b,c, ... = uncertainties.correlated_values(nom_values, corr_mat)`. The IDs don't matter since you aren't concerned with any global state to any other variables.